contracts/wrapperContracts/WrapConversionRate.sol

```diff
@@ -0,0 +1,208 @@
+            // can perform operation.
+            performAddToken();
+        }
+    }
+
+    function performAddToken() internal {
+        conversionRates.addToken(addTokenToken);
+
+        //token control info
+        conversionRates.setTokenControlInfo(
+            addTokenToken,
+            addTokenMinimalResolution,
+            addTokenMaxPerBlockImbalance,
+            addTokenMaxTotalImbalance
+        );
+
+        //step functions
+        int[] memory zeroArr = new int[](1);
+        zeroArr[0] = 0;
+
+        conversionRates.setQtyStepFunction(addTokenToken, zeroArr, zeroArr, zeroArr, zeroArr);
+        conversionRates.setImbalanceStepFunction(addTokenToken, zeroArr, zeroArr, zeroArr, zeroArr);
+
+        conversionRates.enableTokenTrade(addTokenToken);
+    }
+
+    function getAddTokenParameters() public view
+        returns(ERC20 token, uint minimalRecordResolution, uint maxPerBlockImbalance, uint maxTotalImbalance)
+    {
+        token = addTokenToken;
+        minimalRecordResolution = addTokenMinimalResolution;
+        maxPerBlockImbalance = addTokenMaxPerBlockImbalance; // in twei resolution
+        maxTotalImbalance = addTokenMaxTotalImbalance;
+    }
+
+    function getAddTokenSignatures() public view returns (address[] signatures) {
+        uint nonce;
+        (signatures, nonce) = getDataTrackingParameters(addTokenDataIndex);
+        return(signatures);
+    }
+
+    function getAddTokenNonce() public view returns (uint nonce) {
+        address[] memory signatures;
+        (signatures, nonce) = getDataTrackingParameters(addTokenDataIndex);
+        return(nonce);
+    }
+
+    //set token control info
+    ////////////////////////
+    function setTokenInfoData(ERC20 [] tokens, uint[] maxPerBlockImbalanceValues, uint[] maxTotalImbalanceValues)
+        public
+        onlyOperator
+    {
+        require(maxPerBlockImbalanceValues.length == tokens.length);
+        require(maxTotalImbalanceValues.length == tokens.length);
+
+        //update data tracking
+        setNewData(tokenInfoDataIndex);
+
+        tokenInfoTokenList = tokens;
+        tokenInfoPerBlockImbalance = maxPerBlockImbalanceValues;
+        tokenInfoMaxTotalImbalance = maxTotalImbalanceValues;
+    }
+
+    function approveTokenControlInfo(uint nonce) public onlyOperator {
+        if(addSignature(tokenInfoDataIndex, nonce, msg.sender)) {
+            // can perform operation.
+            performSetTokenControlInfo();
+        }
+    }
+
+    function performSetTokenControlInfo() internal {
+        require(tokenInfoTokenList.length == tokenInfoPerBlockImbalance.length);
+        require(tokenInfoTokenList.length == tokenInfoMaxTotalImbalance.length);
+
+        uint minimalRecordResolution;
+        uint rxMaxPerBlockImbalance;
+        uint rxMaxTotalImbalance;
+
+        for (uint i = 0; i < tokenInfoTokenList.length; i++) {
+            (minimalRecordResolution, rxMaxPerBlockImbalance, rxMaxTotalImbalance) =
```

Comment:
btw, `(x,,) = f()` should work in solidity. i.e., you don't have to define the two additional variables